packages/library-base/src/commonMain/kotlin/io/realm/internal/SuspendableWriter.kt

```diff
@@ -53,7 +56,27 @@ internal class SuspendableWriter(private val owner: RealmImpl, val dispatcher: C
         tid = runBlocking(dispatcher) { threadId() }
     }
 
-    suspend fun <R> write(block: MutableRealm.() -> R): Pair<RealmReference, R> {
+    // Currently just for internal-only usage in test, thus API is not polished
+    suspend fun updateSchema(schema: RealmSchemaImpl): FrozenRealmReference {
+        return withContext(dispatcher) {
+            transactionMutex.withLock {
+                realm.log.debug("Updating schema: $schema")
```

Comment:
Does the `toString()` of schema output anything useful? Maybe print path to Realm here?